Fix: Fix colorHex to handle alpha values

laxa88 opened the pull request:
_Issue #, if available:_
- https://github.com/amzn/style-dictionary/issues/362#issuecomment-562206735

_Description of changes:_

By default, when using the `js` transformGroup, all tokens are converted into hex values (for example, `#000000`) before additional transforms are applied. For `rgba` tokens, this causes the color values to lose the transparency value, resulting in all tokens to default to `#xxxxxxFF` (full opacity).

This PR fixes the default transform to use `hex8` to account for RGBA transforms.

I'm not entirely sure if there is a historical reason for using `colorHex` instead of `colorHex8` for `js` transforms, but if this is not an oversight, I would appreciate if someone can share some details. Otherwise, I hope this PR helps. 🙇 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

jorenbroekema commented:
Hi, I think this is mostly an oversight to be honest, I actually would prefer if we didn't have 2 transforms in the first place and just have 1 that handles both hex8 but also uses hex6 when the alpha channel is set 1.
So if possible, can we adjust the regular hex transform to handle hex8 when necessary and deprecate the hex8 specific transform, so we can remove it in the next breaking change but start telling users to use the regular hex one instead?

And can we add some tests to the PR?

laxa88 commented:
@jorenbroekema Thank you for the info and feedback.

- Reverted the change to `colorHex8`
- Updated the `colorHex` logic to handle hex6 and hex8, similar to `colorCss`
- Added and updated `colorHex` tests

jorenbroekema reviewed (approved):
Nice! thanks :)

I added a commit with some documentation / release note stuff to wrap this one up
